python/cudf/cudf/core/dtypes.py

```diff
@@ -166,3 +163,51 @@ def __repr__(self):
             return f"ListDtype({self.element_type.__repr__()})"
         else:
             return f"ListDtype({self.element_type})"
+
+
+class StructDtype(ExtensionDtype):
+
+    name = "struct"
+
+    def __init__(self, fields):
+        """
+        fields : dict
+            A mapping of field names to dtypes
+        """
+        pa_fields = {
+            k: cudf.utils.dtypes.cudf_dtype_to_pa_type(v)
+            for k, v in fields.items()
+        }
+        self._typ = pa.struct(pa_fields)
+
+    @property
+    def fields(self):
+        return {
+            field.name: cudf.utils.dtypes.cudf_dtype_from_pa_type(field.type)
+            for field in self._typ
+        }
+
+    @property
+    def type(self):
+        # TODO: we should change this to return something like a
+        # StructDtypeType, once we figure out what that should look like
+        return dict
```

Comment:
It's supposed to be the scalar type corresponding to this dtype. See [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/dtypes/base.py#L142).

So if we have a series `s` of `StructDtype`,  `s[0]` should return an object of this type. For now `s[0]` doesn't work, but the question is what should we returns when it _does_ work.

Reply:
I think a `StructDtypeType` can be uniquely defined as an ordered list of fields, where each field is made up of a field name (`str`) and a type.

Reply:
Agreed, but if we did that today, users are going to be surprised by that behaviour. They're likely going to expect `s[0]` to return a `dict`, because the equivalent of a `StructColumn` in Pandas is just a Series of dicts..